UI/FleetButton.cpp

```diff
@@ -291,6 +314,40 @@ void FleetButton::LayoutIcons() {
         GG::Pt graphic_ul = middle - subtexture_sz / 2;
         m_selection_indicator->SizeMove(graphic_ul, graphic_ul + subtexture_sz);
     }
+
+    // refresh fleet button tooltip
+    if (m_fleet_blockaded) {
+        std::shared_ptr<Fleet> fleet;
+        std::shared_ptr<System> current_system;
+        std::map<int, bool> adjacent_system_IDs;
+        std::string available_exits = "";
+        int available_exits_count = 0;
+
+        if (!m_fleets.empty())
+            // can just pick first fleet because all fleets in system should have same exits
+            fleet = GetFleet(*m_fleets.begin());
+        else return;
+
+        current_system = GetSystem(fleet->SystemID());
+        adjacent_system_IDs = current_system->StarlanesWormholes();
+
+        for (const auto& target_system_id : adjacent_system_IDs) {
+            if (!fleet->BlockadedAtSystem(fleet->SystemID(), target_system_id.first)) {
```

Comment:
prefer `if (fleet->Blockaded(...)) continue;` so the following can be de-indented